services/github/github-gist-stars.service.js

```diff
@@ -0,0 +1,97 @@
+import gql from 'graphql-tag'
+import Joi from 'joi'
+import { metric } from '../text-formatters.js'
+import { NotFound } from '../index.js'
+import { GithubAuthV4Service } from './github-auth-service.js'
+import { documentation as commonDocumentation } from './github-helpers.js'
+
+const schema = Joi.object({
+  data: Joi.object({
+    viewer: Joi.object({
+      gist: Joi.object({
+        stargazerCount: Joi.number(),
+        url: Joi.string(),
+      }).allow(null),
+    }).required(),
+  }).required(),
+}).required()
+
+const documentation = `${commonDocumentation}
+<p>This badge shows the number of stargazers for a gist. Gist id is accepted as input and 'gist not found' is returned if the gist is not found for the given gist id.
+</p>`
+
+export default class GithubGistStars extends GithubAuthV4Service {
+  static category = 'social'
+
+  static route = {
+    base: 'github/stars/gists',
+    pattern: ':gist',
```

Comment:
Can we make the param `:gistId` to match https://github.com/badges/shields/blob/8e1680d65d12b97356a587db929ecb80e874fd78/services/github/github-gist-last-commit.service.js#L13